docs: add Azure AI Foundry documentation for Claude models

Chesars opened the pull request:
## Title

 docs: add Azure AI Foundry documentation for Claude models

##  Relevant issues

  Fixes #17066

##  Type

  📖 Documentation

  ## Changes

  Add documentation for using Claude models (Sonnet 4.5, Haiku 4.5, Opus 4.1)  deployed on Azure AI Foundry.

  Azure exposes Claude using Anthropic's native /v1/messages API, so users can use the existing anthropic/ provider with their Azure endpoint - no changes to model pricing or new provider needed.

**Documentation includes:**
  - LiteLLM SDK usage example
  - Environment variables configuration
  - LiteLLM Proxy config.yaml example
  - Instructions for finding Azure endpoint

Chesars committed:
docs: add Azure AI Foundry documentation for Claude models

Add documentation explaining how to use Claude models (Sonnet 4.5,
Haiku 4.5, Opus 4.1) deployed on Azure AI Foundry with LiteLLM.

Azure exposes Claude using Anthropic's native API, so users can use
the existing anthropic/ provider with their Azure endpoint.

Closes #17066

krrishdholakia commented on docs/my-website/docs/providers/anthropic.md:
```diff
@@ -181,6 +181,89 @@ Without `LITELLM_ANTHROPIC_DISABLE_URL_SUFFIX`:
 With `LITELLM_ANTHROPIC_DISABLE_URL_SUFFIX=true`:
 - Base URL `https://my-proxy.com/custom/path` → `https://my-proxy.com/custom/path` (unchanged)
 
+### Azure AI Foundry (Microsoft Foundry)
```
i would expect it to be `azure_ai/<claude model>` 

because i assume there might be differences down the road with things like beta headers for azure ai vs. anthropic api - so it would be useful to keep them separate 

cc: @Sameerlite

Chesars committed:
docs: Add alternative method for Azure AI Foundry using anthropic/ provider

Document that users can use anthropic/ provider with Azure endpoint
as an alternative to the dedicated azure_ai/ provider.

jasverma-hub commented:
> Hi @jasverma-hub. This PR adds in the documentation an alternative method to use Anthropic models deployed in Azure, since the Azure API with Anthropic models is almost the same as the direct Anthropic API. In this PR #17202 the integration for azure with Anthropic models was added as a built-in method to use it like `model=f"azure_ai/{deployment_name}",` and adding to .env `AZURE_AI_API_BASE` and `AZURE_AI_API_KEY` . Are you getting any error messages from LiteLLM?

I’ve tried adding the Claude models through both the config.yaml and the LiteLLM UI, but I’m still getting the same routing error. I’m already running the latest LiteLLM version.

This is how I currently have the models defined in my config:

- model_name: claude-sonnet-4-5
    litellm_params:
      model: azure_ai/claude-sonnet-4-5
      api_base: https://######.services.ai.azure.com/anthropic
      api_key: os.environ/OPENAI_API_KEY

- model_name: claude-opus-4-1
    litellm_params:
      model: azure_ai/claude-opus-4-1
      api_base: https://######.services.ai.azure.com/anthropic
      api_key: os.environ/OPENAI_API_KEY

- model_name: claude-opus-4-5
    litellm_params:
      model: azure_ai/claude-opus-4-5
      api_base: https://######.services.ai.azure.com/anthropic
      api_key: os.environ/OPENAI_API_KEY
<img width="1919" height="870" alt="image" src="https://github.com/user-attachments/assets/d0da1fde-8e9b-4207-89f9-6f30347eef9c" />

jasverma-hub commented:
> > > Hi @jasverma-hub. This PR adds in the documentation an alternative method to use Anthropic models deployed in Azure, since the Azure API with Anthropic models is almost the same as the direct Anthropic API. In this PR #17202 the integration for azure with Anthropic models was added as a built-in method to use it like `model=f"azure_ai/{deployment_name}",` and adding to .env `AZURE_AI_API_BASE` and `AZURE_AI_API_KEY` . Are you getting any error messages from LiteLLM?
> > 
> > 
> > I’ve tried adding the Claude models through both the config.yaml and the LiteLLM UI, but I’m still getting the same routing error. I’m already running the latest LiteLLM version.
> > This is how I currently have the models defined in my config:
> > 
> > * model_name: claude-sonnet-4-5
> >   litellm_params:
> >   ```
> >   model: azure_ai/claude-sonnet-4-5
> >   
> >   api_base: https://######.services.ai.azure.com/anthropic
> >   
> >   api_key: os.environ/OPENAI_API_KEY
> >   ```
> > * model_name: claude-opus-4-1
> >   litellm_params:
> >   ```
> >   model: azure_ai/claude-opus-4-1
> >   
> >   api_base: https://######.services.ai.azure.com/anthropic
> >   
> >   api_key: os.environ/OPENAI_API_KEY
> >   ```
> > * model_name: claude-opus-4-5
> >   litellm_params:
> >   ```
> >   model: azure_ai/claude-opus-4-5
> >   
> >   api_base: https://######.services.ai.azure.com/anthropic
> >   
> >   api_key: os.environ/OPENAI_API_KEY
> >   ```
> > 
> > <img alt="image" width="1919" height="870" src="https://private-user-images.githubusercontent.com/81007964/521179100-d0da1fde-8e9b-4207-89f9-6f30347eef9c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ3MDM0MTIsIm5iZiI6MTc2NDcwMzExMiwicGF0aCI6Ii84MTAwNzk2NC81MjExNzkxMDAtZDBkYTFmZGUtOGU5Yi00MjA3LTg5ZjktNmYzMDM0N2VlZjljLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEyMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMjAyVDE5MTgzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc1ZWMyYTEyMTYwNDdkYzgyMmJkMTE3ODMxYjNkODI5ZjZiNmNmZTE2OTU3Mzk1ZmVmM2IyMmFjN2YxODA3MTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ngg_3EG2NpEkDGQXP0ihhaYCsM49mE_OXF9e3j49KhI">
> 
> The feature for `model=azure_ai/claude-sonnet-4-5` should be available in the next release.


> Hi @jasverma-hub. This PR adds in the documentation an alternative method to use Anthropic models deployed in Azure, since the Azure API with Anthropic models is almost the same as the direct Anthropic API. In this PR #17202 the integration for azure with Anthropic models was added as a built-in method to use it like `model=f"azure_ai/{deployment_name}",` and adding to .env `AZURE_AI_API_BASE` and `AZURE_AI_API_KEY` . Are you getting any error messages from LiteLLM?

How can I add this to the config.yaml? Will it work if I follow the instructions in the Anthropic documentation on LiteLLM?

Chesars commented:
The other option is to use the alternative method documented in this PR. https://github.com/BerriAI/litellm/pull/17104/files.

Chesars commented:
Thanks for reporting it, we'll look into that. So yes, we'll have to wait for the next release.

jasverma-hub commented:
Okay, fine. One last thing: is the problem with Claude models or Claude models hosted by Azure?
